### DISCUSSION TITLE
Can't run dApp in production (build)

### DISCUSSION THREAD

**Discussion by raffle-x-dev** [rainbow-me/rainbowkit#1730] [Q&A]
Hi there.
I'm trying to make a production build and test it locally with http-server.
I run npm run build, I get this, seeming to be ok

`✓ Linting and checking validity of types    
   Creating an optimized production build ...
 ✓ Compiled successfully
 ✓ Collecting page data    
 ✓ Generating static pages (3/3) 
 ✓ Collecting build traces    
 ✓ Finalizing page optimization    
But:
1st: I can't see any "build" folder created.
2nd: if I navigate to .next/server/pages/ where the static files are created and try to test from there, the console log tells me it doesn't find the realtive .css and stuff just created.

`ed3b855a4261c72f.css:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)
0e19a8a03c0e9127.css:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)
webpack-3d7e0abfabeb4ddb.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)
framework-314c182fa7e2bf37.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)
main-e9a4e4ae37deaffb.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)
_app-7921d25ab76bd16c.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)
index-55cfce5acd66dc77.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)
_buildManifest.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)
_ssgManifest.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 (Not Found)`

How do I build for production in RainbowKit? :/ TY
![image](https://github.com/rainbow-me/rainbowkit/assets/157182235/9acfbc39-7a34-450d-8b4b-5580b69ead06)

![image](https://github.com/rainbow-me/rainbowkit/assets/157182235/d791073f-8bf0-45ab-a329-d4de6a787307)

**Comment by raffle-x-dev** [comment]
Sorry for the noob argument, but couldn't find much in the docs.

I also tried
`npm run build`
and then
`npm run start`
and localhost:3000 shows a working product. Is this the way to test production builds locally? Why the other way doesn't work and how shall I deploy to live environment if those files can't find each other?

**Reply by 0xmkh** [reply]
If you do build and start that's correct for next.js. If you can just link to a small reproducible example that'd be the best.

**Reply by raffle-x-dev** [reply]
Yeah maybe `npm run start` is the way to go, but what about deplyoing online to a proper website? Which files am I gonna need to pick up if there's no build folder?

**Reply by 0xmkh** [reply]
The build folder is being generated after `npm run build`. In next.js the build folder is `.next` so you can use that and point the build folder to `.next`.

**Comment by 0xmkh** [comment]
@raffle-x-dev  Can you link us a small reproducible example ? Also what rainbowkit and wagmi version are you using ?

**Reply by raffle-x-dev** [reply]
` "dependencies": {
    "@openzeppelin/contracts": "^5.0.1",
    "@rainbow-me/rainbowkit": "^1.3.3",
    "next": "^14.0.4",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.21.3",
    "viem": "~1.21.4",
    "wagmi": "~1.4.13"
  },
  "devDependencies": {
    "@types/node": "^18.19.3",
    "@types/react": "^18.2.43",
    "eslint": "^8.15.0",
    "eslint-config-next": "^14.0.4",
    "typescript": "^5.0.4"
  },`

**Reply by 0xmkh** [reply]
Awesome are you able to link us the code as well ? CodeSandbox or a github repo code is also good
